google/cloud/bigtable/examples/bigtable_examples_common_test.cc

```diff
@@ -0,0 +1,143 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/bigtable/examples/bigtable_examples_common.h"
+#include "google/cloud/testing_util/scoped_environment.h"
+#include <gmock/gmock.h>
+
+namespace google {
+namespace cloud {
+namespace bigtable {
+namespace examples {
+
+TEST(BigtableExamplesCommon, Simple) {
+  int test_calls = 0;
+  Example example({
+      {"test",
+       [&](std::vector<std::string> const& args) {
+         ++test_calls;
+         if (args.empty()) throw Usage("test-usage");
+         ASSERT_EQ(2, args.size());
+         EXPECT_EQ("a0", args[0]);
+         EXPECT_EQ("a1", args[1]);
+       }},
+  });
+  char argv0[] = "argv0";
```

Comment:
Perhaps `Run()` should take a `char const* argv[]`?

Reply:
I am planning to pass the arguments from `main()` directly into this function, so that would not work, right?

Reply:
I believe that if you change `Run(int argc, char* argv[])` to `Run(int argc, char const* const* argv)` that you be able to ...
```
  char const* args[] = {"1", "2", "3"};
  Run(3, args);
```
and
```
int main(int argc, char* argv[]) {
  Run(argc, argv);
}
```

Reply:
I will send a follow up PR to fix that.